parser/Parser/CIFF.h

```diff
@@ -75,8 +75,19 @@ class CIFF {
 		uint64_t tags_size = header_size - sizeof(magic) - sizeof(header_size) - sizeof(content_size) - sizeof(width) - sizeof(height) - caption.size() - 1;
 		readTags(r, tags_size);
 
+
+		if (header_size != tags_size + sizeof(magic) + sizeof(header_size) + sizeof(content_size) +
+			sizeof(width) + sizeof(height) + caption.length() + 1) {
+			throw ParsingException();
+		}
+		uint64_t characters_read_counter = r.getBytesRead();
 		pixels.resize(width*height);
 		Pixel* firstElem = &(pixels[0]);
 		r.readArray(firstElem, width*height);
+		uint64_t characters_read_counter_after = r.getBytesRead();
+
+		if (content_size != characters_read_counter_after - characters_read_counter) {
+			throw ParsingException();
+		}
```

Comment:
```suggestion
		uint64_t size_read = sizeof(magic) + sizeof(header_size) + sizeof(content_size) + sizeof(width) + sizeof(height) + caption.size() + 1;
		if ((uint64_t)header_size < size_read) {
			throw ParsingException();
		}
		uint64_t tags_size = header_size - size_read;
		readTags(r, tags_size);

		uint64_t characters_read_counter = r.getBytesRead();
		pixels.resize(width*height);
		Pixel* firstElem = &(pixels[0]);
		r.readArray(firstElem, width*height);
		uint64_t characters_read_counter_after = r.getBytesRead();
```

I think we do not need these two checks, but one other might be helpful.

1) There is already a check in place for `width*height*3 == content_size` here: https://github.com/radl97/BiztonsagosGang-2021/pull/1/files/1cf36830f97946be5dc9969be20e649e9062b882..75d4ee6d9542cdf6c572b20a2cbb60c49753e4d1#diff-7475ab861b14f6af32da0169dd75cb430ad297f7b295cf414efdd18fc22d752bR65
This means that only failing read could get problematic. But Reader throws an exception in that case, so we are good.

2) `header_size`-related check is not needed, as we calculate the `tags_size` from the `header_size` already. However, there is an integer overflow error in my code, which is fixed here.

If you are still not sure whether this works as intended, then PM me :D Do not commit this change too soon :) (even if the deadline is approaching)